[git-webkit] Look for hooks in the GIT_DIR, not the worktree

emw-apple opened the pull request:
#### f05f3c94288477d8b1350ab85f44cb21223b20ed
<pre>
[git-webkit] Look for hooks in the GIT_DIR, not the worktree
<a href="https://bugs.webkit.org/show_bug.cgi?id=254677">https://bugs.webkit.org/show_bug.cgi?id=254677</a>

Reviewed by Jonathan Bedard.

Currently, git-webkit assumes hooks are in a &quot;.git/hooks&quot; directory
relative to the top-level worktree directory. This is not the case for
repos with multiple worktrees or other exotic setups.

Use common_dir when computing the path to the hooks directory.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_hooks.py:
(InstallHooks.hook_needs_update):

Canonical link: <a href="https://commits.webkit.org/262288@main">https://commits.webkit.org/262288@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/660bdd245bee5efa3a749c5977a7e96ac796e6f3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1136 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1167 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1205 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1837 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1000 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1120 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1218 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1247 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1193 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1144 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1056 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1049 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1717 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/1170 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1092 "Build is in progress. Recent messages:OS: Monterey (12.6), Xcode: 14.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests (warnings)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1047 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1035 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1037 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1076 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/2127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1084 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/1123 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1023 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1056 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/280 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1105 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
Worst part is that we totally _write_ hooks into the right location in this exact same file.

emw-apple committed:
[git-webkit] Look for hooks in the GIT_DIR, not the worktree
https://bugs.webkit.org/show_bug.cgi?id=254677

Reviewed by Jonathan Bedard.

Currently, git-webkit assumes hooks are in a ".git/hooks" directory
relative to the top-level worktree directory. This is not the case for
repos with multiple worktrees or other exotic setups.

Use common_dir when computing the path to the hooks directory.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_hooks.py:
(InstallHooks.hook_needs_update):

Canonical link: https://commits.webkit.org/262288@main
